query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -250,6 +255,30 @@ private String convertOperator2String(Operator operator) {
     }
   }
 
+  //called for selection columns
+  private String convertExpression2String(Expression expression, Params.Builder paramsBuilder,int selectionCounter) {
+    switch (expression.getValueCase()) {
+      case COLUMNIDENTIFIER:
+        String logicalColumnName = expression.getColumnIdentifier().getColumnName();
+        // this takes care of the Map Type where it's split into 2 columns
+        List<String> columnNames = viewDefinition.getPhysicalColumnNames(logicalColumnName);
+        return joiner.join(columnNames);
+      case LITERAL:
+        return convertLiteralToString(expression.getLiteral(), paramsBuilder);
+      case FUNCTION:
+        selectionMap.put(expression.getFunction().getFunctionName(),selectionCounter);
```

Comment:
If the key of the selection map is function name, 
1. then wouldn't it be overwritten if there are multiple occurrence of the same function name 
2. and if multiple occurrences of a function is not possible in the query then why use a map

Reply:
Yes. I guess map signature should be changed too map<Integer,String> from map<String,Integer>

Reply:
> Hashmap is fast for retrieving <key,value> data?

yes it is.

Moving to <integer, string> should fix it. 

However, I am having doubts on returning this map itself